pytorch_lightning/core/lightning.py

```diff
@@ -1794,11 +1799,39 @@ def to_onnx(
 
         input_sample = self._apply_batch_transfer_handler(input_sample)
 
+        if not isinstance(input_sample, (Tuple, list)):
+            input_sample = (input_sample, )
+
         if "example_outputs" not in kwargs:
             self.eval()
-            kwargs["example_outputs"] = self(input_sample)
+            kwargs["example_outputs"] = self(*input_sample)
 
         torch.onnx.export(self, input_sample, file_path, **kwargs)
+
+        if model_check is None:
+
+            def to_numpy(tensor):
+                return tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
+
+            def model_check(file_path, input_sample, torch_outs):
+
+                # Generic graph integrity checks
+                onnx_model = onnx.load(file_path)
+                onnx.checker.check_model(onnx_model)
+
+                # Get ONNX outputs
+                ort_session = onnxruntime.InferenceSession(file_path)
+                ort_inputs = {inp.name: to_numpy(sample) for inp, sample in zip(ort_session.get_inputs(), input_sample)}
+                ort_outs = ort_session.run(None, ort_inputs)
+
+                # Compare against PyTorch outputs
+                if not isinstance(torch_outs, (Tuple, list)):
+                    torch_outs = (torch_outs, )
+                for ort_out, torch_out in zip(ort_outs, torch_outs):
+                    np.testing.assert_allclose(to_numpy(torch_out), ort_out, rtol=1e-03, atol=1e-05)
```

Comment:
you can save an indentation level by having the check function definitions first and pushing the if to the end.